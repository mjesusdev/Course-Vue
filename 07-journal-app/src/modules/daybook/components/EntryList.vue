<template>
  <h1>EntryList</h1>
</template>

<script>
    export default {

    }
</script>

<style>

</style>