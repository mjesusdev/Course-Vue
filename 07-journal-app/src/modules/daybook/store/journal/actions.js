/* export const myAction = async ({ commit }) => {

} */

export const loadEntries = async() => {
    
}

export const updateEntry = async() => {

}

export const createEntry = async() => {

}