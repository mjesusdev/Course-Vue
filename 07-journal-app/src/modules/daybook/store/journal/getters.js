/* export const myGetter = ( state ) => {
    return state
} */

export const getEntriesByTerm = ( state ) => ( term = '' ) => {
    if ( term.length === 0 )  return state.entries

    return state.entries.filter( entry => entry.text.toLowerCase().includes( term.toLocaleLowerCase() ) )
}

export const getEntryById = () => {
    
}