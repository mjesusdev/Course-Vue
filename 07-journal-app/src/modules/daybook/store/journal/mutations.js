/* export const myAction = ( state ) => {

} */

export const setEntries = () => {

}

export const updateEntry = () => {

}

export const addEntry = () => {

}