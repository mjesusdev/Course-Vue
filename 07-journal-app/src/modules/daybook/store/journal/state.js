export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(), // 123456789
            date: new Date().toDateString(), // thursday 21 October
            text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Porro ea accusantium, sequi saepe quis obcaecati pariatur aut, molestiae dolor in rerum sit distinctio maiores omnis asperiores tempora laudantium! Nemo, ullam.',
            picture: null // https
        },
        {
            id: new Date().getTime() + 1000, // 1234567893232
            date: new Date().toDateString(), // thursday 21 October
            text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Porro ea accusantium, sequi saepe quis obcaecati pariatur aut, molestiae dolor in rerum sit distinctio maiores omnis asperiores tempora laudantium! Nemo, ullam.',
            picture: null // https
        },
        {
            id: new Date().getTime() + 2000, // 1234567893232234
            date: new Date().toDateString(), // thursday 21 October
            text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Porro ea accusantium, sequi saepe quis obcaecati pariatur aut, molestiae dolor in rerum sit distinctio maiores omnis asperiores tempora laudantium! Nemo, ullam.',
            picture: null // https
        }
    ]
})