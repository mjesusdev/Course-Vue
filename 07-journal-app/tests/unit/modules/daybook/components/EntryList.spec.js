import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'

import journal from '@/modules/daybook/store/journal'
import EntryList from '@/modules/daybook/components/EntryList'
import { journalState } from '../../../mock-data/test-journal-state'

const createVuexStore = ( initialState ) => 
createStore({
    modules: {
        journal: {
            ...journal,
            state: { ...initialState }
        }

    }
})

describe('Tests in EntryList', () => {

    const store = createVuexStore( journalState )
    const mockRouter = {
        push: jest.fn()
    }

    let wrapper

    beforeEach(() => {
        jest.clearAllMocks()

        wrapper = shallowMount( EntryList , {
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        })
    })

    test('should call getEntriesByTerm without term and show 3 entries', () => {
        expect( wrapper.findAll('entry-stub').length ).toBe(3)
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('should call getEntriesByTerm and filter entries', async() => {
        const input = wrapper.find('input')
        await input.setValue('DayBook')

        expect( wrapper.findAll('entry-stub').length ).toBe(1)
    })

    test('button should redirect to /new', () => {
        wrapper.find('button').trigger('click')

        expect( mockRouter.push )
            .toHaveBeenCalledWith({ name: 'entry', params: { id: 'new' } })
    })

})