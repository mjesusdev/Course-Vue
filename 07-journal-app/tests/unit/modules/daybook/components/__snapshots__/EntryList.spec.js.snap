// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in EntryList should call getEntriesByTerm without term and show 3 entries 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entrada"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3 mb-2"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
